### ISSUE TITLE
Code syntax colorizer can allow choosing separate colours when using onenote in dark mode.

### ISSUE THREAD

**Issue by Rishabhsahu325** [stevencohn/OneMore#1158]
Wehn using onenote in dark mode and adding colorizer to certain code , I noticed that a lot of code content is coloured black . So it does not legible among the dark coloured background. So when applying the colorizer ,it will be useful if color scheme of colorizer is different for dark mode.
Current workaround for this would probably be to shift to light mode temporarily ,then apply colorizer then shift to dark mode. But it will be less convenient as compared to copying text in vscode or online IDE and then copying the syntax highlighted text back from that IDE. 
I don't think it is a bug or a new feature, just an improvement to the pre existing feature.

**Comment by stevencohn** [comment]
OneMore has different colors for light and dark so this may be a bug. Which language are you using?

**Comment by stevencohn** [comment]
@Rishabhsahu325 Can you show me an example of what you mean please?

**Comment by Rishabhsahu325** [comment]
Ok ,so here's an example with c++ code
![Screenshot 2023-11-19 115730](https://github.com/stevencohn/OneMore/assets/17985200/c103f3aa-1bcb-4707-b195-055d76463fd3)

**Comment by stevencohn** [comment]
@Rishabhsahu325  Sorry but which version of OneMore are you running. This should have been fixed in [release 5.9](https://github.com/stevencohn/OneMore/releases/tag/5.9.0) for #1083

**Comment by Rishabhsahu325** [comment]
![image](https://github.com/stevencohn/OneMore/assets/17985200/9f2dfe50-fae7-4bbc-8987-c1525b14073c)
These are the versions
Version 5.9.0 with OneNote 16.0.16924.20150

**Comment by stevencohn** [comment]
Then I'm having trouble reproducing the issue with 5.9.0. I've set the background to "dark" using the View/Switch Background button and also tried setting it using the OneMore Set Page Color command. Both work as expected for C# and C++. It looks like you're using C++.

I presume you're selecting the block of code and choosing Colorize/C++. Is there anything else you're doing differently?

**Comment by stevencohn** [comment]
Note also that if you colorize and _then_ switch the background, you then need to recolorize the code. It will not automatically switch the default text color. 

Change the background or Switch Background and _then_ colorize.

This is a known limitation. The only way to make this appear automatic is to define a color palette that, in my opinion, just doesn't look good in either mode.

**Comment by Rishabhsahu325** [comment]
I did not use the View /Switch Background for dark mode . I used the Account settings option that sets everything to dark mode. 
![image](https://github.com/stevencohn/OneMore/assets/17985200/ad61bcca-b2d6-4fde-b144-17892efa119c)

Other than that just as you said ,I used the Colorizer/C++.

**Comment by stevencohn** [comment]
That's effectively the same.

**Comment by Rishabhsahu325** [comment]
The difference is that when I open onenote ,dark mode is default in my case ,but in your case I believe that I had to click the switch background option each time. And since you are not able to reproduce the issue ,I think this difference may be related to it

**Comment by stevencohn** [comment]
Nope the switch background is persisted when you close OneNote. It's the same; I normally run in Dark mode but switched the background to White. Either way, it affects the same Registry setting.

So if you start from scratch with a dark page, paste some code, and then colorize it, you're still seeing black text?

**Comment by stevencohn** [comment]
What Windows theme are you using? If it's not the default dark mode theme, then maybe that's it?

**Comment by Rishabhsahu325** [comment]
The screenshot which I sent you had the dark blue page from the beginning in the dark mode . And since I have selected the dark mode under account settings the view menu switch option by default even if it persists ,will change content to light mode on clicking once and dark mode on clicking twice to deselect it.

**Comment by Rishabhsahu325** [comment]
![image](https://github.com/stevencohn/OneMore/assets/17985200/a29693d5-6d5c-4420-b146-2b97f6e580e0)
Do you mean this?  
Otherwise my windows theme is set to dark mode and all start menus /context menus appear in the shade of the settings screen above.

**Comment by Rishabhsahu325** [comment]
Hi , so I just remembered about this issue and came to check here. I tried the version 6.1.0 and saw that the behavior is the same. Should we mark it as a bug?

**Comment by stevencohn** [comment]
No. I'll take another look at this ticket. need to refresh my memory....

**Comment by stevencohn** [comment]
Here's what I tried.

0. Windows theme is set to Dark mode by default, highlight color is brown
1. OneNote theme is set to system default (which means dark/black mode)
2. With OneNote page background switched to white/light, given this page

<img width="506" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/b1c9268a-f759-4281-9111-ebabd47bf8e2">

3. Colorize as C++

<img width="533" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/56d973e0-1cc5-4b26-97a0-6ff42f242a97">

4. Ctrl+Z to undo colorize, switch background to dark

<img width="509" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/c2c5ac0a-606a-4a19-b2b4-d4f8fa04b894">

5. Colorize as C++

<img width="530" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/e6eda842-3cf0-4157-9af0-88bb273098f1">

6. Ctrl+Z to undo colorize, switch backgroudn to light.
7. Use OneMore's Set Page Color to change background to a dark color (here dark cyan)

<img width="483" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/d5ea7fd9-e900-4366-8a0a-eda16684489f">

8. Colorize as C++

<img width="529" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/aabbb8ae-5017-4db0-9111-bdce9ba9b5d4">

9. Ctrl+Z to undo colorize, reset page color to white
10. Switch background to dark, Set Page Color to dark cyan

<img width="353" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/11371ec5-4a74-40c9-9cd9-2c29067a7f04">

<img width="485" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/81401090-8809-413c-82da-83179459849c">

11. Colorize as C++

<img width="528" alt="image" src="https://github.com/stevencohn/OneMore/assets/9964757/b4c1a650-5b8c-4141-8306-96db8c7c06e9">

Here you can see that if we have Switched Background to dark mode AND chosen to set the background of the page to a dark color (which is actually an inverted light color, hard to explain but true) then the default text color is dark on a dark background and hard to read. But all other test cases are OK.

So unless you've both switched to dark background mode AND chosen a dark page color, it should work as designed in OneMore 6.1.0

**Comment by stevencohn** [comment]
Fixed for next release.

|OneNote Mode | OneMore Set Page Color | Tested|
| -- | -- | -- |
| light | auto | OK |
| light | light  | OK |
| light | dark  | OK |
| dark | auto | OK |
| dark | light  | OK |
| dark | dark  | OK |
